### PR TITLE
Remove size check from testutil assertor

### PR DESCRIPTION
**Pull request by black-adder**

### CODE REVIEW

File: metrics/testutils/testutils.go [hunk 1]
```diff
@@ -17,19 +17,33 @@ type ExpectedMetric struct {
 
 // TODO do something similar for Timers
 
+// AssertCounterMetric checks if counter metric exists.
+func AssertCounterMetric(t *testing.T, f *metrics.LocalFactory, expectedMetrics ExpectedMetric) {
```
Comment (yurishkuro) [inline metrics/testutils/testutils.go]: Why not do `expectedMetrics ...ExpectedMetric`?

File: metrics/testutils/testutils.go [hunk 2]
```diff
@@ -17,19 +17,33 @@ type ExpectedMetric struct {
...
+}
+
+// AssertGaugeMetric checks if gauge metric exists.
+func AssertGaugeMetric(t *testing.T, f *metrics.LocalFactory, expectedMetrics ...ExpectedMetric) {
+	_, gauges := f.Snapshot()
+	assertMetrics(t, gauges, expectedMetrics...)
+}
+
+// AssertCounterMetrics checks if the existing counter metrics exactly match the expected metrics.
 func AssertCounterMetrics(t *testing.T, f *metrics.LocalFactory, expectedMetrics []ExpectedMetric) {
```
Comment (yurishkuro) [inline metrics/testutils/testutils.go]: we don't need this method if we have L21

Comment (black-adder) [inline metrics/testutils/testutils.go]: this function checks the size as well, ie exact match versus do metrics exist in the set

Comment (black-adder) [inline metrics/testutils/testutils.go]: I guess I can refactor it to call the above func then do the size check

Comment (yurishkuro) [inline metrics/testutils/testutils.go]: yeah, we should do that, but I would then rename the first one AssertCounterMetrics (plural) and the second AssertCounterMetricAndSize. Ideally, I'd prefer to minimize the API surface - how important is the size check?

Comment (black-adder) [inline metrics/testutils/testutils.go]: the size check is an artifact that I carried over from go-client, and looks superfluous now looking at it

**Pull request merged**